src/lib/util/common.d.ts

```diff
@@ -0,0 +1,76 @@
+import type { Queue, Song } from '@lavaclient/queue';
+import type { ActionRowBuilder, AttachmentBuilder, AutocompleteInteraction, Client, Collection, EmbedBuilder, MessageActionRowComponentBuilder, Snowflake, TextChannel, VoiceChannel } from 'discord.js';
+import type { Node, Player } from 'lavaclient';
+import type PlayerHandler from '../PlayerHandler.js';
+import type ReplyHandler from '../ReplyHandler.js';
+
+export type SearchStateRecord = {
+    pages: { info: Song }[][];
+    timeout: ReturnType<typeof setTimeout>;
+    selected: Snowflake[];
+};
+
+export type SettingsPageOptions = 'language' | 'format';
+
+export type SettingsPage = {
+    current: string;
+    embeds: EmbedBuilder[];
+    actionRow: ActionRowBuilder;
+};
+
+export type MessageOptionsBuilderInputs = string | EmbedBuilder | (string | EmbedBuilder)[];
+
+export type MessageOptionsBuilderOptions = {
+	type?: 'success' | 'neutral' | 'warning' | 'error';
+	components?: ActionRowBuilder<MessageActionRowComponentBuilder>[];
+	files?: AttachmentBuilder[];
+};
```

Comment:
```suggestion
export type MessageOptionsBuilderOptions = {
	type?: 'success' | 'neutral' | 'warning' | 'error';
    force?: 'reply' | 'edit' | 'update';
} & AdditionalBuilderOptions
```

Reply:
additional is not always part of the base

Reply:
Right, there was `PlayerHandler` so that makes sense.